server/controllers/contest.js

```diff
@@ -0,0 +1,95 @@
+const asyncHandler = require("express-async-handler");
+const Contest = require('../models/Contest')
+
+// handler for creating a new contest object
+exports.createContest = asyncHandler(async (req, res, next) => {
+    const { title, description, user, prizeAmount, deadline } = req.body;
+
+    const contest = await Contest.create({ title, description, prizeAmount, deadline });
+
+    if (contest) {
+        res.status(201).json({
+            success: {
+                contest: {
+                    created: contest.created,
+                    title: contest.title,
+                    description: contest.description
+                }
+            }
+        });
+    } else {
+        res.status(400);
+        throw new Error("invalid contest data");
+    }
+
+    next()
+
+});
+
+// handler for updating a contest object
+exports.updateContest = asyncHandler(async (req, res, next) => {
+
+    try {
+        const id = req.params.id
```

Comment:
You can declare these variables (33-36) outside of the try/catch block and also destructure the `req.body`  :)

Reply:
Done.